app/Topology.hs

```diff
@@ -25,23 +28,44 @@ data TopologyInfo = TopologyInfo {
 --
 -- TODO: this type should be extended to take into account IPv6 addresses.
 --
-data NodeAddress = NodeAddress HostName ServiceName
+data NodeAddress = NodeAddress {
+      naHostName :: HostName
+    , naPort     :: PortNumber
+    }
   deriving (Eq, Ord, Show)
 
-nodeAddressToSockAddr :: NodeAddress -> SockAddr
-nodeAddressToSockAddr (NodeAddress addr port) = SockAddrInet (read port) (IP.toHostAddress (read addr))
-
 instance Condense NodeAddress where
-    condense (NodeAddress addr port) = addr ++ ":" ++ show port
+    condense NodeAddress {naHostName, naPort} = naHostName ++ ":" ++ show naPort
 
 instance FromJSON NodeAddress where
     parseJSON = withObject "NodeAddress" $ \v -> NodeAddress
-      <$> v .: "addr"
-      <*> v .: "port"
+      <$> v .: "hostName"
```

Comment:
I think that `hostName` may be confusing since we a nodes should be given an IP address to bind to, not a domain name. What about `hostAddress`?

Reply:
ok, I'll fix this.

Reply:
Fixed